### PR TITLE
fix: passing x-forwarded-host in axios request

### PR DESCRIPTION
**Pull request by Fifciu**
Closed and moved to: https://github.com/vuestorefront/vue-storefront/pull/6869

### CODE REVIEW

File: packages/core/src/utils/nuxt/_proxyUtils.ts [hunk 1]
```diff
@@ -38,7 +38,10 @@ export const getIntegrationConfig = (context: NuxtContext, configuration: any) =
       baseURL: new URL(/\/api\//gi.test(baseURL) ? '' : 'api', baseURL).toString(),
       headers: {
         ...(cookie ? { cookie } : {}),
-        ...(context.req ? { Host: context.req.headers.host } : {})
+        ...(context.req ? {
+          'X-Forwarded-Host': context.req.headers['x-forwarded-host'],
```
[the remark this answers is not in the document]
Comment (Fifciu) [reply packages/core/src/utils/nuxt/_proxyUtils.ts:42]: We would have `'X-Forwarded-Host': undefined` provided as an argument to the lodash's `merge` which will remove this property: `Source properties that resolve to undefined are skipped if a destination value exists`

[the remark this answers is not in the document]
Comment (bartoszherba) [reply packages/core/src/utils/nuxt/_proxyUtils.ts:42]: 👌

**Review by bartoszherba — APPROVED** [review]

File: packages/core/src/utils/nuxt/_proxyUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fifciu) [reply packages/core/src/utils/nuxt/_proxyUtils.ts]: You know what... it works here well in unit test without mocking `merge`. But when I applied the same in EPCC it didn't work. Maybe it's caused be some versions mismatch. IDK. But I decided to add x-forwarded-host only if it's present

File: packages/core/src/utils/nuxt/_proxyUtils.ts [hunk 2]
```diff
@@ -38,7 +38,12 @@ export const getIntegrationConfig = (context: NuxtContext, configuration: any) =
       baseURL: new URL(/\/api\//gi.test(baseURL) ? '' : 'api', baseURL).toString(),
       headers: {
         ...(cookie ? { cookie } : {}),
-        ...(context.req ? { Host: context.req.headers.host } : {})
+        ...(context.req ? {
```
Comment (lsliwaradioluz) [inline packages/core/src/utils/nuxt/_proxyUtils.ts:41]: These ternaries with destructuring terrify me. I think we could make it more readable... It's hard to tell what is going on there at first sight.

Approved, but please, consider my comment.

**Review by lsliwaradioluz — APPROVED** [review]

File: packages/core/src/utils/nuxt/_proxyUtils.ts (hunk 2 above)
Comment (jeffpdotone) [reply packages/core/src/utils/nuxt/_proxyUtils.ts:41]: Could we do it like this instead?

```typescript
 const resolvedHost = context?.req?.headers?.['x-forwarded-host'] || context?.req?.headers?.host;

  return merge({
    axios: {
      baseURL: new URL(/\/api\//gi.test(baseURL) ? '' : 'api', baseURL).toString(),
      headers: {
        ...(cookie ? { cookie } : {}),
        ...(resolvedHost ? { Host: resolvedHost } : {})
      }
    }
  }, configuration);
```

And as I received comment from @WojtekTheWebDev in [PR#677](https://github.com/vuestorefront/bigcommerce/pull/677) we don't need 'x-forwarded-host' and needed to ensure Host is there coming from either x-forwarded-host or directly from host property of request headers.

Comment (Fifciu) [reply packages/core/src/utils/nuxt/_proxyUtils.ts]: Well @jeffpdotone, I don't really like the idea of merging two headers into one. If we have 2 headers let's leave them as they are during forwarding or we will end up having confusing code. To be honest, that was the primary goal of this PR.

Comment (jeffpdotone) [reply packages/core/src/utils/nuxt/_proxyUtils.ts]: All good then if the purpose is served and fixed the main issue.

**Review by jeffpdotone — APPROVED** [review]
All good just need to conflict fix.

**Review by lsliwaradioluz — APPROVED** [review]
Now better!

**Pull request closed without merging**